persist/sqlite/hosts.go

```diff
@@ -65,115 +65,152 @@ func (s *Store) HostsForScanning(minLastAnnouncement time.Time, limit uint64) (r
+		}
+		if params.MaxContractPrice != nil {
+			filters = append(filters, "CASE WHEN v2=1 THEN rhp4_prices_contract_price <= ? ELSE settings_contract_price <= ? END")
+			args = append(args, encode(*params.MaxContractPrice), encode(*params.MaxContractPrice))
+		}
+		if params.MaxUploadPrice != nil {
+			filters = append(filters, "CASE WHEN v2=1 THEN rhp4_prices_ingress_price <= ? ELSE settings_upload_bandwidth_price <= ? END")
+			args = append(args, encode(*params.MaxUploadPrice), encode(*params.MaxUploadPrice))
+		}
+		if params.MaxDownloadPrice != nil {
+			filters = append(filters, "CASE WHEN v2=1 THEN rhp4_prices_egress_price <= ? ELSE settings_download_bandwidth_price <= ? END")
+			args = append(args, encode(*params.MaxDownloadPrice), encode(*params.MaxDownloadPrice))
+		}
+		if params.MaxBaseRPCPrice != nil {
+			filters = append(filters, "settings_base_rpc_price <= ?")
+			args = append(args, encode(*params.MaxBaseRPCPrice))
+		}
+		if params.MaxSectorAccessPrice != nil {
+			filters = append(filters, "settings_sector_access_price <= ?")
+			args = append(args, encode(*params.MaxSectorAccessPrice))
+		}
+		if params.AcceptContracts != nil {
+			v := 0
+			if *params.AcceptContracts {
+				v = 1
+			}
+			filters = append(filters, fmt.Sprintf("CASE WHEN v2=1 THEN rhp4_settings_accepting_contracts = %d ELSE settings_accepting_contracts = %d END", v, v))
+		}
+		if params.Online != nil {
+			v := 0
+			if *params.Online {
+				v = 1
+			}
+			filters = append(filters, fmt.Sprintf("last_scan_successful = %d", v))
+		}
+		args = append(args, limit, offset)
+
+		var sortColumn string
+		switch sortBy {
+		case explorer.HostSortDateCreated:
+			sortColumn = "known_since"
+		case explorer.HostSortNetAddress:
+			sortColumn = "net_address"
+		case explorer.HostSortPublicKey:
+			sortColumn = "public_key"
+		case explorer.HostSortUptime:
+			sortColumn = uptimeValue
+		case explorer.HostSortAcceptingContracts:
+			sortColumn = "CASE WHEN v2=1 THEN rhp4_settings_accepting_contracts ELSE settings_accepting_contracts END"
+		case explorer.HostSortStoragePrice:
+			sortColumn = "CASE WHEN v2=1 THEN rhp4_prices_storage_price ELSE settings_storage_price END"
+		case explorer.HostSortContractPrice:
+			sortColumn = "CASE WHEN v2=1 THEN rhp4_prices_contract_price ELSE settings_contract_price END"
+		case explorer.HostSortDownloadPrice:
+			sortColumn = "CASE WHEN v2=1 THEN rhp4_prices_egress_price ELSE settings_download_bandwidth_price END"
+		case explorer.HostSortUploadPrice:
+			sortColumn = "CASE WHEN v2=1 THEN rhp4_prices_ingress_price ELSE settings_upload_bandwidth_price END"
+		case explorer.HostSortUsedStorage:
+			sortColumn = "CASE WHEN v2=1 THEN rhp4_settings_used_storage ELSE settings_used_storage END"
+		case explorer.HostSortTotalStorage:
+			sortColumn = "CASE WHEN v2=1 THEN rhp4_settings_total_storage ELSE settings_total_storage END"
+		default:
+			return fmt.Errorf("invalid sort column: %s", sortBy)
```

Comment:
This should probably be an exported error we can check for to properly return a 400 status code when it happens.